util/iterutil/slices.go

```diff
@@ -0,0 +1,25 @@
+package iterators
+
+import "iter"
+
+// SlicePointers returns an iterator that yields indices and pointers to the elements of a slice.
+func SlicePointers[Slice ~[]T, T any](s Slice) iter.Seq2[int, *T] {
+	return func(yield func(int, *T) bool) {
+		for i := range s {
+			if !yield(i, &s[i]) {
+				break
```

Comment:
Is there a reason you are using `break` here instead of `return`. It seems that both should give the same result. I may be missing something subtle, but otherwise it may be nice to have a consistent pattern in these functions.

Reply:
DONE